2d/CMakeLists.txt

```diff
@@ -29,8 +29,6 @@ set(impl_incs
   "include/pcl/${SUBSYS_NAME}/impl/morphology.hpp"
 )
 
-include_directories("${CMAKE_CURRENT_SOURCE_DIR}/include")
```

Comment:
This module does not use `PCL_ADD_LIBRARY`, neither does geometry. Can this line still be removed?

Reply:
It did work due to the include_directories from within PCL_SUBSYS_DEPEND - I have removed it now and changed to use interface libraries for the header only modules.

Reply:
I removed the last changes, as it seemlingly only work like this, with the GPU and CUDA submodules. I couldn't "fix" it by adding include_directories in their respective submodules, which was propergated correctly to the other dependencies.
So my reasoning is to keep the include_directories, until the latest PR which removes these and fix the remaining issues that aren't present until they are actually removed.
Want me to split the PRs even further or can we merge these 4 step by step?